packages/@aws-cdk/aws-codeartifact/lib/repository.ts

```diff
@@ -0,0 +1,404 @@
+ */
+export abstract class RepositoryBase extends Resource implements IRepository {
+
+  /** @attribute */
+  public abstract readonly repositoryArn: string = '';
+  /** @attribute */
+  public abstract readonly repositoryName: string = '';
+  /** @attribute */
+  public abstract readonly repositoryDescription: string = '';
+  /** @attribute */
+  public abstract readonly repositoryDomainOwner: string = '';
+  /** @attribute */
+  public abstract readonly repositoryDomainName: string = '';
+
+  constructor(scope: Construct, id: string) {
+    super(scope, id, {});
+  }
+
+  /** Implement the {@link IRepository.grantRead} method. */
+  public grantRead(identity: iam.IGrantable): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee: identity,
+      actions: [...actions.readActions],
+      resourceArns: [this.repositoryArn],
+    });
+  }
+
+  /** Implement the {@link IRepository.grantWrite} method. */
+  public grantWrite(identity: iam.IGrantable): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee: identity,
+      actions: [...actions.writeActions],
+      resourceArns: [this.repositoryArn],
+    });
+  }
+
+  /** Implement the {@link IRepository.grantReadWrite} method. */
+  public grantReadWrite(identity: iam.IGrantable): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee: identity,
+      actions: [...actions.readActions, ...actions.writeActions],
+      resourceArns: [this.repositoryArn],
+    });
+  }
+
+  /**
+     * Defines a CloudWatch event rule which triggers for repository events. Use
+     * `rule.addEventPattern(pattern)` to specify a filter.
+     */
+  public onEvent(id: string, options: events.OnEventOptions = {}) {
+    const rule = new events.Rule(this, id, options);
+    rule.addEventPattern({
+      source: ['aws.codeartifact'],
+      detail: {
+        domainName: [this.repositoryDomainName],
+        domainOwner: [this.repositoryDomainOwner],
+        repositoryName: [this.repositoryName],
+      },
+    });
+    rule.addTarget(options.target);
+    return rule;
+  }
+
+  /**
+     * Defines a CloudWatch event rule which triggers when a "CodeArtifact Package
+     *  Version State Change" event occurs.
+     */
+  public onPackageVersionStateChange(id: string, options: events.OnEventOptions = {}): events.Rule {
+    const rule = this.onEvent(id, options);
+    rule.addEventPattern({
+      detailType: ['CodeArtifact Package Version State Change'],
+    });
+    return rule;
+  }
+}
+
+/**
+ * A new CodeArtifacft repository
+ */
+export class Repository extends RepositoryBase {
+  /**
+     * Import an existing Repository provided an ARN
+     *
+     * @param scope The parent creating construct
+     * @param id The construct's name
+     * @param repositoryArn repository ARN (i.e. arn:aws:codeartifact:us-east-2:444455556666:repository/my-domain/my-repo)
+     */
+  public static fromRepositoryArn(scope: Construct, id: string, repositoryArn: string): IRepository {
```

Comment:
Similar comments to `fromDomainArn`; I think this will need to be removed in place of `fromRespositoryAttributes` that takes in the arn, domain, etc.